modules/mailer/mailer.go

```diff
@@ -45,151 +44,141 @@ func NewMessage(to []string, subject, body string) *Message {
 			hostname, err = os.Hostname()
 			if err != nil {
-				return err
+				return nil, err
 			}
 		}
-
-		if err = client.Hello(hostname); err != nil {
-			return fmt.Errorf("Hello: %v", err)
-		}
+		dialer.LocalName = hostname
+	} else {
+		dialer.LocalName = ""
 	}
 
-	// If not using SMTPS, alway use STARTTLS if available
-	hasStartTLS, _ := client.Extension("STARTTLS")
-	if !isSecureConn && hasStartTLS {
-		if err = client.StartTLS(tlsconfig); err != nil {
-			return fmt.Errorf("StartTLS: %v", err)
+	if opts.UseCertificate {
+		cert, err := tls.LoadX509KeyPair(opts.CertFile, opts.KeyFile)
+		if err != nil {
+			return nil, err
 		}
+		dialer.TLSConfig.Certificates = []tls.Certificate{cert}
 	}
 
-	canAuth, options := client.Extension("AUTH")
-	if canAuth && len(opts.User) > 0 {
-		var auth smtp.Auth
-
-		if strings.Contains(options, "CRAM-MD5") {
-			auth = smtp.CRAMMD5Auth(opts.User, opts.Passwd)
-		} else if strings.Contains(options, "PLAIN") {
-			auth = smtp.PlainAuth("", opts.User, opts.Passwd, host)
-		} else if strings.Contains(options, "LOGIN") {
-			// Patch for AUTH LOGIN
-			auth = LoginAuth(opts.User, opts.Passwd)
-		}
+	return dialer, nil;
+}
 
-		if auth != nil {
-			if err = client.Auth(auth); err != nil {
-				return fmt.Errorf("Auth: %v", err)
-			}
-		}
+func Test(opts *setting.Mailer) error {
+	dialer, err := newDialer(opts)
+	if err != nil {
+		return err
 	}
 
-	if err = client.Mail(from); err != nil {
-		return fmt.Errorf("Mail: %v", err)
+	log.Debug("Mailer: Dialing %s", opts.Host)
+	conn, err := dialer.Dial()
+	if err != nil {
+		if terr, ok := err.(*textproto.Error); ok {
+			terr.Msg = "Failed to connect: " + terr.Msg
+		}
+		return err
+	} else {
+		conn.Close()
 	}
+	return nil
+}
 
-	for _, rec := range to {
-		if err = client.Rcpt(rec); err != nil {
-			return fmt.Errorf("Rcpt: %v", err)
-		}
+func Send(msg *Message) error {
+	opts := setting.MailService
+
+	dialer, err := newDialer(opts)
+	if err != nil {
+		return err
 	}
 
-	w, err := client.Data()
+	log.Debug("Mailer: Dialing %s", opts.Host)
+	conn, err := dialer.Dial()
 	if err != nil {
-		return fmt.Errorf("Data: %v", err)
-	} else if _, err = msg.WriteTo(w); err != nil {
-		return fmt.Errorf("WriteTo: %v", err)
-	} else if err = w.Close(); err != nil {
-		return fmt.Errorf("Close: %v", err)
+		if terr, ok := err.(*textproto.Error); ok {
+			terr.Msg = "Failed to connect: " + terr.Msg
+		}
+		return err
 	}
+	defer conn.Close()
 
-	return client.Quit()
+	if err := conn.Send(opts.From, msg.GetHeader("To"), msg.Message); err != nil {
+		return err
+	} else {
+		log.Trace("E-mail sent %s: %s", msg.GetHeader("To"), msg.Info)
+	}
+	return nil
 }
 
 func processMailQueue() {
-	sender := &Sender{}
+	opts := setting.MailService
+
+	dialer, err := newDialer(opts)
+	if err != nil {
+		return
+	}
+	open := false
+	var conn gomail.SendCloser
 
 	for {
 		select {
+		// A new message is pending
 		case msg := <-mailQueue:
+			// Dial SMTP if neccessary
+			if !open {
```

Comment:
Yes, that's true.
But in my opinion it is an unfriendly behavior of a client to leave the connection open for minutes and thus consuming resources on server and client side without a real need.